kafka-plugins-common/src/main/java/io/cdap/plugin/batch/source/KafkaBatchConfig.java

```diff
@@ -190,74 +192,85 @@ public Schema getMessageSchema() {
       // if the field is not the time field and not the key field, it is a message field.
       if (fieldName.equals(keyField)) {
         if (fieldType != Schema.Type.BYTES) {
-          throw new IllegalArgumentException("The key field must be of type bytes or nullable bytes.");
+          collector.addFailure("The key field must be of type bytes or nullable bytes.", null);
```

Comment:
```
.withConfigProperty("keyField")
.withOutputSchemaField(keyField)
```

This should highlight both the keyField textbox as well as the corresponding field in the output schema.